<launch>
  <group ns="stereo_preprocessing">
  <!-- stereo preprocessing -->
  <arg name="INPUT_POINT_CLOUD" default="/multisense/resize_1_4/points" />
  <!-- compute normal -->
  <node pkg="jsk_topic_tools" type="standalone_complexed_nodelet"
        name="preprocessing_manager" output="screen" respawn="true">
    <rosparam subst_value="true">
      nodelets:
        - name: relay
          type: jsk_topic_tools/Relay
          remappings:
            - from: ~input
              to: $(arg INPUT_POINT_CLOUD)
        - name: normal_estimation
          type: jsk_pcl/NormalEstimationIntegralImage
          remappings:
            - from: ~input
              to: relay/output
        - name: stereo_ground_cloud
          type: jsk_pcl/TfTransformCloud
          remappings:
            - from: ~input
              to: relay/output
        - name: stereo_downsampled_ground_cloud
          type: jsk_topic_tools/Relay
          remappings:
            - from: ~input
              to: stereo_ground_cloud/output
    </rosparam>
  </node>
  <group ns="stereo_ground_cloud">
    <rosparam>
      target_frame_id: ground
    </rosparam>
  </group>
  </group>
</launch>
